Fix type hints in ansi2html/converter.py

After adding type hints in #143 there are now four type hinting errors that are currently ignored. They need to be fixed.

```
ansi2html/converter.py:422: error: Incompatible types in "yield" (actual type "Type[CursorMoveUp]", expected type "Union[str, CursorMoveUp]")
ansi2html/converter.py:524: error: Unsupported right operand type for in ("Union[str, OSC_Link]")
ansi2html/converter.py:570: error: Incompatible return value type (got "Union[bool, str, Set[str]]", expected "str")
ansi2html/converter.py:578: error: Unsupported right operand type for in ("Union[bool, str, Set[str]]")
Found 4 errors in 1 file (checked 6 source files)
```

--- a/ansi2html/converter.py
+++ b/ansi2html/converter.py
@@ -531,7 +531,12 @@ class Ansi2HTMLConverter:
                 if final_parts:
                     final_parts.pop()
 
-                while final_parts and "\n" not in final_parts[-1]:
+                while final_parts and (
+                    isinstance(final_parts[-1], OSC_Link)
+                    or (
+                        isinstance(final_parts[-1], str) and "\n" not in final_parts[-1]
+                    )
+                ):
                     final_parts.pop()
 
                 continue

